python-spec/src/somacore/query/query.py

```diff
@@ -757,14 +757,18 @@ class AxisIndexer:
     def _obs_index(self) -> pd.Index:
         """Private. Return an index for the ``obs`` axis."""
         if self._cached_obs is None:
-            self._cached_obs = pd.Index(data=self.query.obs_joinids().to_numpy())
+            self._cached_obs = self._index_factory(
+                data=self.query.obs_joinids().to_numpy()
```

Comment:
the mypy error here is because it’s expecting the argument to be positional; removing `data=` should do it.